src/commands/deploy.js

```diff
@@ -29,15 +29,20 @@ class DeployCommand extends AbstractAuthenticatedCommand {
    * Get selected environment; prompt for environment when none is provided.
    * @param {Object} config - Actual command config (including command parameters).
    * @throws Will throw an error when there is no environment (unreachable).
-   * @throws Will throw an error when there is no production environment.
+   * @throws Will throw an error when there is no reference environment.
    * @return {Object} The environment found.
    */
   async getEnvironment(config) {
     const environments = await new EnvironmentManager(config).listEnvironments();
     const productionExists = environments.find((environment) => environment.type === 'production');
```

Comment:
`productionExists` have no place anymore here
I think you can drop entirely this section, so getEnvironment will be:

`const environmentName = config.ENVIRONMENT_NAME
      || await this.selectEnvironment(environments);
    return environments.find((environment) => environment.name === environmentName);`

Reply:
what do you this of that?

Reply:
We can discuss how to enhance `listEnvironments` to exclude the reference in another scope

Reply:
IMO I will split `listEnvironments` into:
 - `listDeployableEnvironments`  for command deploy
 - `listEnvironmentToPushOnto` for command push
 - `listAllEnvironments` for environment:list command

Reply:
I agree, will do the change